content/js/record.md

```diff
@@ -1,307 +1,331 @@
+```
 
-### Deleting Referenced Record
+If some delete operation fails (e.g. due to insufficient permission) the error
+message(s) will be reflected in the errors array illustrated above.
 
-Yet to be implemented. For now, you have to delete the referencing record
-first and then the referenced record.
 
 <a name="data-type"></a>
-## Data Type
+## Data Types
+
+Skygear supports most of the built-in JavaScript types:
 
-Skygear supports almost all of the builtin JavaScript types, such as:
 - String
 - Number
 - Boolean
 - Array
 - Object
 - Date
 
-There are also four other types provided by Skygear JS SDK:
-- Reference (described above)
-- Sequence (described below)
-- Geolocation (described in [Geo-location](/js/guide/geolocation) section)
-- Asset (described in [Assets (File Upload)](/js/guide/asset) section)
+In addition, Skygear JS SDK 4 more types:
+
+- [Reference](#reference)
+- [Sequence](#sequence)
+- [Geo Location](/js/guide/geolocation)
+- [Assets](/js/guide/asset) (File Uploads)
+
+Refer to the [server](/server/guide/data-type) documentation for
+more detail on supported data types.
+
+
+<a name="reference"></a>
+## Records Relations
+
+Skygear supports relation between records via _references_.
+`skygear.Reference` is a class which will translate to foreign keys in
+the skygear database for efficient queries.
+
+``` javascript
+const Message = skygear.Record.extend('message');
+const Person = skygear.Record.extend('person');
+
+const bob = new Person({ name: 'bob' });
+const alice = new Person({ name: 'alice' });
+const message  = new Message({
+  from: new skygear.Reference(bob),
+  to: new skygear.Reference(alice),
+  content: 'foo bar baz qux'
+});
+
+skygear.publicDB.save([bob, alice, message]);
+
+```
+
+**Note:** Since reference objects translates into foreign key constraints, the order of which records are saved matters here, saving `message` before either `bob` or `alice` will result
+in an error. Also note that you must delete the `message` record before
+`bob` or `alice`.
 
-Please refer to the [server](/server/guide/data-type) documentation for
-more detail in supported data types.
+Using references, you can retrieve the referenced records in a single query using
+[transient includes](/js/guide/query#relational-queries).
 
-### Auto-Incrementing Sequence Fields
 
-Skygear reserves the `id` field in the top level of all record as a primary key.
-`id` must be unique and default to be Version 4 UUID. If you want to
-auto-incrementing id for display purpose, Skygear provide a sequence for this 
-purpose. The sequence is guaranteed unique. Once a record with sequence id is
-saved, all the other records will automatically have sequence ids as well.
+<a name="sequence"></a>
+## Auto-Incrementing Sequences
+
+All skygear records uses a UUIDv4 `id` as the primary key in the database,
+you can add a sequence field if you want a sequential "id"
+for each record, this field is guaranteed to be unique.
```

Comment:
We need to mention that the `noteID` does not need to be included in subsequent record creation - It will be automatically generated.